new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -138,36 +146,69 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
     const logEventNumRef = useRef(logEventNum);
     const numPagesRef = useRef<number>(STATE_DEFAULT.numPages);
     const pageNumRef = useRef<Nullable<number>>(STATE_DEFAULT.pageNum);
+    const receivedNumChunksRef = useRef<number>(0);
+    const logExportManagerRef = useRef<LogExportManager>(new LogExportManager());
 
     const mainWorkerRef = useRef<null|Worker>(null);
 
+    logExportManagerRef.current.setNumChunks(Math.ceil(numEvents / EXPORT_LOGS_CHUNK_SIZE));
```

Comment:
This could be error-prone. 
Depending on the interface changes to be made, this might no longer be necessary.